lib/credentials/msiTokenCredentials.ts

```
@@ -26,38 +27,34 @@ export interface MSITokenResponse {
  * This object can only be used to acquire token on a virtual machine provisioned in Azure with managed service identity.
  */
 export class MSITokenCredentials {
+  private static readonly defaultPort = 50342;
+  private static readonly defaultResource = "https://management.azure.com/";
+
+  port: number;
+  resource: string;
+
   public constructor(
     /**
-     * @property {string} domain - The domain or tenant id for which the token is required.
-     */
-    public domain: string,
-    /**
-     * @property {number} port - Port on which the MSI service is running on the host VM. Default port is 50342
-     */
-    public port = 50342,
-    /**
-     * @property {string} resource - The resource uri or token audience for which the token is needed.
-     * For e.g. it can be:
-     * - resourcemanagement endpoint "https://management.azure.com"(default)
-     * - management endpoint "https://management.core.windows.net/"
-     */
-    public resource = "https://management.azure.com",
-    /**
-     * @property {string} aadEndpoint - The add endpoint for authentication. default - "https://login.microsoftonline.com"
+     * @property {LoginWithMSIOptions} options - Optional parameters
      */
-    public aadEndpoint = "https://login.microsoftonline.com") {
-    if (!Boolean(domain) || typeof domain.valueOf() !== "string") {
-      throw new TypeError("domain must be a non empty string.");
-    }
-    if (typeof port.valueOf() !== "number") {
+    public options: MSIOptions) {
+
+    if (!options) options = {};
+
+    if (!options.port) {
```

You may want to check for `options.port == undefined` instead of `!options.port`. I know in this case it doesn't matter because a port of 0 is invalid, but at least a consumer of this package would fail quickly and learn that they were providing a port of 0 instead of thinking that 0 converted to the default port.